source/common/config/metadata.h

```diff
@@ -116,8 +116,20 @@ template <typename factoryClass> class TypedMetadataImpl : public TypedMetadata
    */
   void populateFrom(const envoy::config::core::v3::Metadata& metadata) {
     auto& data_by_key = metadata.filter_metadata();
+    auto& typed_data_by_key = metadata.typed_filter_metadata();
     for (const auto& [factory_name, factory] :
          Registry::FactoryRegistry<factoryClass>::factories()) {
+      const auto& typed_meta_iter = typed_data_by_key.find(factory_name);
+      // If the key exists in Any metadata, and parse() not return nullptr,
+      // populate data_.
+      if (typed_meta_iter != typed_data_by_key.end()) {
+        auto result = factory->parse(typed_meta_iter->second);
+        if (result != nullptr) {
```

Comment:
I think it's unclear from the description of the `parse()` method that it should return `nullptr` if one doesn't want to implement it.

Reply:
Yes, make sense. Modified the comments to explicitly state that.